src/transformime/transformers.ts

```diff
@@ -66,24 +101,40 @@ class JavascriptTransformer implements ITransformer<Widget> {
   }
 }
 
+
+/**
+ * A transformer for `<svg>` data.
+ */
 export
 class SVGTransformer implements ITransformer<Widget> {
   mimetypes = ['image/svg+xml'];
   transform(mimetype: string, data: string): Widget {
     let w = new Widget();
     w.node.innerHTML = data;
+    let svgElement = w.node.getElementsByTagName('svg')[0];
+    if (!svgElement) {
+      throw new Error("SVGTransform: Error: Failed to create <svg> element");
+    }
     return w;
   }
 }
 
+
+/**
+ * A transformer for LateX data.
+ */
 export
 class LatexTransformer implements ITransformer<Widget> {
   mimetypes = ['text/latex'];
   transform(mimetype: string, data: string): Widget {
     let w = new Widget();
     w.node.textContent = data;
-    // TODO: do something to the class to get MathJax to render it?
-    // TODO: use katex if mathjax isn't defined?
+    if (typeof MathJax !== "undefined") {
+      w.node.textContent = data;
+      MathJax.Hub.Queue(["Typeset", MathJax.Hub, w.node]);
```

Comment:
onAfterShow is called when isVisible toggles from false to true independent
of attachment.
On Mar 23, 2016 15:23, "Jason Grout" <notifications@github.com> wrote:

> In src/transformime/transformers.ts
> <https://github.com/jasongrout/jupyter-js-notebook/pull/1#discussion_r57230043>
> :
>
> >  export
> >  class LatexTransformer implements ITransformer<Widget> {
> >    mimetypes = ['text/latex'];
> >    transform(mimetype: string, data: string): Widget {
> >      let w = new Widget();
> >      w.node.textContent = data;
> > -    // TODO: do something to the class to get MathJax to render it?
> > -    // TODO: use katex if mathjax isn't defined?
> > +    if (typeof MathJax !== "undefined") {
> > +      w.node.textContent = data;
> > +      MathJax.Hub.Queue(["Typeset", MathJax.Hub, w.node]);
>
> Digging a bit deeper, it does seem that MathJax deals with the node being display:
> none by making a hidden node and measuring text on that. It's not ideal,
> but it is a workaround for this case.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jasongrout/jupyter-js-notebook/pull/1/files/fbe00611b2c11fb89efec73a74dca9698b34247e#r57230043>
>